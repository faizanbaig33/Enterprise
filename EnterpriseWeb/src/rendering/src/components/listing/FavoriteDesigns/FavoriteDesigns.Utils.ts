import { ButtonGroupProps } from 'src/helpers/ButtonGroup';
import { ImageToggleWrapperProps } from 'src/helpers/ImageToggleWrapper/ImageToggleWrapper';

export const getImageToggleData = (props: any): ImageToggleWrapperProps => {
  return {
    fields: {
      primaryImage: {
        value: {
          src: props.interiorImage.src,
          width: 387,
          height: 387,
        },
      },
      primaryImageMobile: {
        value: {
          src: props.interiorImage.src,
          width: 387,
          height: 387,
        },
      },
      secondaryImage: {
        value: {
          src: props.exteriorImage.src,
          width: 387,
          height: 387,
        },
      },
      secondaryImageMobile: {
        value: {
          src: props.exteriorImage.src,
          width: 387,
          height: 387,
        },
      },
    },
  };
};

export const getButtonGroupData = (props: any): Partial<ButtonGroupProps> => {
  return {
    fields: {
      cta1Style: {
        name: '',
        id: '',
        url: '',
        fields: {
          Value: {
            value: 'primary',
          },
        },
      },
      cta1Link: {
        value: {
          href: props.requestAQuoteUrl,
          linktype: 'internal',
          text: 'Get a quote',
          querystring: '',
          target: '',
          id: '{7FB335D2-8E99-458E-9EF9-562A78CCB821}',
        },
      },
      cta1Icon: {
        id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
        url: '',
        name: 'Arrow',
        displayName: 'Arrow',
        fields: {
          Value: {
            value: 'arrow',
          },
        },
        templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
        templateName: 'Enum',
      },
      cta1ModalLinkText: { value: '' },
      cta2Style: {
        name: '',
        url: '',
        id: '',
        fields: {
          Value: {
            value: 'link',
          },
        },
      },
      cta2Link: {
        value: {
          href: '/',
          linktype: 'internal',
          text: 'Edit design selections',
          querystring: '',
          target: '',
          id: '{7FB335D2-8E99-458E-9EF9-562A78CCB821}',
        },
      },
      cta2Icon: {
        id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
        url: '',
        name: 'Arrow',
        displayName: 'Arrow',
        fields: {
          Value: {
            value: 'arrow',
          },
        },
        templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
        templateName: 'Enum',
      },
      cta2ModalLinkText: { value: '' },
    },
  };
};
