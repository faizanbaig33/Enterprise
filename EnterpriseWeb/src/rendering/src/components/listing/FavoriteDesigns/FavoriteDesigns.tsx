// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
// Components
import { Component } from 'src/helpers/Component';
// import { FavoriteDesignsTheme } from './FavoriteDesigns.theme';
import { getLocalStorageItem } from 'lib/utils/client-storage-utils';
import { useEffect, useRef, useState } from 'react';
import { Headline } from 'src/helpers/Headline';
import FavoriteDesignCard from './FavoriteDesignCard';
import { useExternalScript } from 'lib/utils/use-external-script';
import { SvgIcon } from 'src/helpers/SvgIcon';
import { hashCode } from 'src/helpers/Component/Component';
import { DesignProps } from './FavoriteDesigns.Types';

export type FavoriteDesignsProps =
  Feature.EnterpriseWeb.Components.Listing.Favorites.FavoriteDesigns;

const FavoriteDesigns = (props: FavoriteDesignsProps) => {
  const [designs, setDesigns] = useState<Array<DesignProps>>([]);
  const componentRef = useRef(null);

  const externalScript = 'https://static.addtoany.com/menu/page.js';
  const state = useExternalScript(externalScript);

  const { fields } = props;

  useEffect(() => {
    setDesigns(getLocalStorageItem('aw_favoritedesigns') as Array<DesignProps>);
  }, []);

  useEffect(() => {
    if (window.a2a_config) {
      window.a2a_config.onclick = 1;
      window.a2a_config.num_services = 8;
      window.a2a_config.templates = {
        email: {
          subject: 'Check out my Favorites from Andersen Windows',
          body: 'Link to my Favorites on AndersenWindows.com ${link}',
        },
        sms: {
          body: 'Link to my Favorites on AndersenWindows.com ${link}',
        },
      };
    }
  }, [state]);

  const printFavoriteDesigns = () => {
    const printContent = document.getElementById(
      fields?.sectionId?.value || `id${hashCode(props.rendering.dataSource)}`
    );
    const originalContents = document.body.innerHTML;

    if (printContent) {
      const notToBePrinted = printContent.querySelector('#print-icon');
      if (notToBePrinted) {
        notToBePrinted.remove();
      }

      document.body.innerHTML = printContent.innerHTML;
      window.print();

      document.body.innerHTML = originalContents;
    }
  };

  const getCreatedAtDate = (dateString: string): string => {
    console.log(dateString);
    const date = new Date(dateString);
    const locale = 'en-us';
    const weekday = date.toLocaleDateString(locale, { weekday: 'long' });
    const month = date.toLocaleDateString(locale, { month: 'long' });
    const day = date.toLocaleDateString(locale, { day: '2-digit' });
    const year = date.toLocaleDateString(locale, { year: 'numeric' });

    return `Created ${weekday} ${month}, ${day} ${year}`;
  };

  if (!fields) return null;

  return (
    <Component variant="lg" dataComponent="general/favoritedesigns" {...props}>
      <div className="col-span-12">
        <div className="flex justify-between">
          <Headline {...props} />
          <div
            id="print-icon"
            className="hidden cursor-pointer items-center justify-center text-primary md:flex"
            onClick={printFavoriteDesigns}
          >
            <SvgIcon icon="print" />
            <span className="ml-xxs">Print</span>
          </div>
        </div>
      </div>
      <div className="col-span-12" ref={componentRef}>
        {designs &&
          designs.length > 0 &&
          designs.map((designData) => {
            return (
              <>
                <span className="font-serif text-small">
                  {getCreatedAtDate(designData.createdDate)}
                </span>
                <div className="relative border border-gray">
                  <FavoriteDesignCard {...designData} />
                </div>
              </>
            );
          })}
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<FavoriteDesignsProps>(FavoriteDesigns);
