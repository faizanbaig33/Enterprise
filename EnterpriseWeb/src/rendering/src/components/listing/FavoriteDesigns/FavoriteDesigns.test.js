// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
import 'lib/jest/match-media-mock';
// Local
import FavoriteDesigns from './FavoriteDesigns';
import defaultData from './FavoriteDesigns.mock-data';

it('renders correctly', () => {
  const component = snapshot(FavoriteDesigns, { componentProps: defaultData });
  hasDataComponent(component, 'general/favoritedesigns');
});
