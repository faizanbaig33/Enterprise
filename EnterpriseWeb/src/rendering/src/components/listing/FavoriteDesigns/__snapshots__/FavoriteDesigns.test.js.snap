// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="general/favoritedesigns"
  id="id1695242607"
>
  <div
    class="my-8 lg:mx-auto md:max-w-screen-lg theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s"
    >
      <div
        class="col-span-12"
      >
        <div
          class="flex justify-between"
        >
          <div
            class="items-top font-sans  text-theme-text text-sm-m md:text-m font-bold mb-s"
          >
            <h3>
              Favorite Designs
            </h3>
            
          </div>
          <div
            class="hidden cursor-pointer items-center justify-center text-primary md:flex"
            id="print-icon"
          >
            <span />
            <span
              class="ml-xxs"
            >
              Print
            </span>
          </div>
        </div>
      </div>
      <div
        class="col-span-12"
      />
    </div>
  </div>
</section>
`;
